lib/sycamore/sycamore/docset.py

```diff
@@ -987,6 +1011,144 @@ def llm_query(self, query_agent: LLMTextQueryAgent, **kwargs) -> "DocSet":
         queries = LLMQuery(self.plan, query_agent=query_agent, **kwargs)
         return DocSet(self.context, queries)
 
+    def top_k(
+        self,
+        llm: LLM,
+        field: str,
+        k: Optional[int],
```

Comment:
Why is k optional? Is the idea that without k specified, all values are returned? Maybe include this in the docstring.